tests/integration/workflows/test_workflows_tasks_upload.py

```diff
@@ -213,324 +202,539 @@ def test_regression_store_record_does_not_commit_when_error(workflow_app):
             )
             with pytest.raises(ValueError):
                 store_record(workflow, eng)
 
 
+@patch(
+    "inspirehep.modules.workflows.tasks.classifier.json_api_request",
+    side_effect=fake_classifier_api_request,
+)
+@patch(
+    "inspirehep.modules.workflows.tasks.magpie.json_api_request",
+    side_effect=fake_magpie_api_request,
+)
+def test_store_record_inspirehep_api_literature_update_not_marked_as_update(
+    mocked_api_request_magpie, mocked_api_request_classifier, workflow_app
+):
+    record_data = {
+        "workflow_data": {
+            "$schema": "http://localhost:5000/schemas/records/hep.json",
+            "_collections": ["Literature"],
+            "document_type": ["article"],
+            "titles": [{"title": "Superconductivity"}],
+            "acquisition_source": {"source": "arXiv"},
+            "dois": [{"value": "10.3847/2041-8213/aa9110"}],
+        },
+        "extra_data": {
+            "source_data": {
+                "data": {
+                    "$schema": "http://localhost:5000/schemas/records/hep.json",
+                    "_collections": ["Literature"],
+                    "document_type": ["article"],
+                    "titles": [{"title": "Superconductivity"}],
+                    "acquisition_source": {"source": "arXiv"},
+                    "dois": [{"value": "10.3847/2041-8213/aa9110"}],
+                },
+                "extra_data": {},
+                "persistent_data": {"marks": {"restart-count": 0}},
+            },
+        },
+    }
+
+    workflow = workflow_object_class.create(record_data)
+    obj = build_workflow(data_type="hep", **record_data)
+    wf_id = obj.id
+    workflow.extra_data["is-update"] = False
+
+    eng = MagicMock(workflow_definition=MagicMock(data_type="hep"))
+    with patch.dict(
+        workflow_app.config,
+        {
+            "FEATURE_FLAG_ENABLE_REST_RECORD_MANAGEMENT": True,
+            "INSPIREHEP_URL": "http://web:8000",
+        },
+    ):
+        with requests_mock.Mocker() as requests_mocker:
+            requests_mocker.register_uri(
+                "POST",
+                "{url}/literature".format(
+                    url=workflow_app.config.get("INSPIREHEP_URL")
+                ),
+                [
+                    {
+                        "status_code": 400,
+                        "json": {
+                            "status": 400,
+                            "message": "PIDAlreadyExists: pid_type:'arxiv', pid_value:'2211.15599'.",
+                        },
+                        "headers": {"content-type": "application/json"},
+                    }
+                ],
+            )
+
+            store_record(obj, eng)  # not throwing exception
+            wf = workflow_object_class.get(wf_id)
+            assert (
+                wf.extra_data["source_data"]["persistent_data"]["marks"][
+                    "restart-count"
+                ]
+                == 1
+            )
+
+
```

Comment:
@drjova here is the check for the restart count that is increased during the restart. If we add another `increase_restart_count` it the `store_record` we'll end up with a doubled number of restarts.